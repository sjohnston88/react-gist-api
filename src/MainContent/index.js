import React, {Component} from 'react';
import Gist from '../Gist/index';
import DefaultMessage from '../DefaultMessage/index'

class MainContent extends Component {
  
  render(){
    return (
      <main> 
        <div className="GistPreview">
          {this.props.invalidDate ? <p><b>Error:</b> Please enter a valid ISO 8601 date</p> : null }
          {this.props.userName === 'defunkt' ? <DefaultMessage /> : null}
          
          {this.props.dataFound ? 
            <div><h2>Viewing Gists by {this.props.userName}</h2><Gist className="GistPreview" id={this.props.gistPreview} /></div>
            : <div><h1><span role="img" aria-label="hmm emoji">ðŸ¤”</span> Nothing found for user "{this.props.userName}"</h1><p>Try entering another username in the address bar.</p></div>
          }
        </div>
      </main>
    );
  }
}

export default MainContent;