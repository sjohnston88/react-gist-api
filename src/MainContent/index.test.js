import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import MainContent from './index';

Enzyme.configure({ adapter: new Adapter() });

describe('Component/MainContent', () => {
  const setup = () => shallow(<MainContent invalidDate={true} userName="defunkt" dataFound={true} />);

  it('renders correctly', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot('default');
  });

  it('renders a error message if the date is invalid', () => {
    const wrapper = setup();
    wrapper.setState({ invalidDate: true });
    expect(wrapper).toMatchSnapshot('invalidDate');
  });

  it('it renders the default message', () => {
    const wrapper = setup();
    wrapper.setState({ userName: 'Defunkt' });
    expect(wrapper).toMatchSnapshot('defaultMessage');
  });

  it('it renders the Gist if data is found', () => {
    const wrapper = setup();
    wrapper.setState({ dataFound: true });
    expect(wrapper).toMatchSnapshot('dataFound');
  });
  
});
