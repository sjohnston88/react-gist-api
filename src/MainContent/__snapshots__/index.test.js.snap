// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dataFound 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainContent
    dataFound={true}
    invalidDate={true}
    userName="defunkt"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="GistPreview"
>
        <p>
                <b>
                        Error:
                </b>
                 Please enter a valid ISO 8601 date
        </p>
        <DefaultMessage />
        <div>
                <h2>
                        Viewing Gists by 
                        defunkt
                </h2>
                <Gist
                        className="GistPreview"
                        id={undefined}
                />
        </div>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            <b>
                        Error:
            </b>
             Please enter a valid ISO 8601 date
</p>,
          <DefaultMessage />,
          <div>
            <h2>
                        Viewing Gists by 
                        defunkt
            </h2>
            <Gist
                        className="GistPreview"
                        id={undefined}
            />
</div>,
        ],
        "className": "GistPreview",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <b>
                Error:
</b>,
              " Please enter a valid ISO 8601 date",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Error:",
              },
              "ref": null,
              "rendered": "Error:",
              "type": "b",
            },
            " Please enter a valid ISO 8601 date",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Viewing Gists by 
                defunkt
</h2>,
              <Gist
                className="GistPreview"
                id={undefined}
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Viewing Gists by ",
                  "defunkt",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Viewing Gists by ",
                "defunkt",
              ],
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "GistPreview",
                "id": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="GistPreview"
>
          <p>
                    <b>
                              Error:
                    </b>
                     Please enter a valid ISO 8601 date
          </p>
          <DefaultMessage />
          <div>
                    <h2>
                              Viewing Gists by 
                              defunkt
                    </h2>
                    <Gist
                              className="GistPreview"
                              id={undefined}
                    />
          </div>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              <b>
                            Error:
              </b>
               Please enter a valid ISO 8601 date
</p>,
            <DefaultMessage />,
            <div>
              <h2>
                            Viewing Gists by 
                            defunkt
              </h2>
              <Gist
                            className="GistPreview"
                            id={undefined}
              />
</div>,
          ],
          "className": "GistPreview",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <b>
                  Error:
</b>,
                " Please enter a valid ISO 8601 date",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Error:",
                },
                "ref": null,
                "rendered": "Error:",
                "type": "b",
              },
              " Please enter a valid ISO 8601 date",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2>
                  Viewing Gists by 
                  defunkt
</h2>,
                <Gist
                  className="GistPreview"
                  id={undefined}
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Viewing Gists by ",
                    "defunkt",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Viewing Gists by ",
                  "defunkt",
                ],
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "GistPreview",
                  "id": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`default 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainContent
    dataFound={true}
    invalidDate={true}
    userName="defunkt"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="GistPreview"
>
        <p>
                <b>
                        Error:
                </b>
                 Please enter a valid ISO 8601 date
        </p>
        <DefaultMessage />
        <div>
                <h2>
                        Viewing Gists by 
                        defunkt
                </h2>
                <Gist
                        className="GistPreview"
                        id={undefined}
                />
        </div>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            <b>
                        Error:
            </b>
             Please enter a valid ISO 8601 date
</p>,
          <DefaultMessage />,
          <div>
            <h2>
                        Viewing Gists by 
                        defunkt
            </h2>
            <Gist
                        className="GistPreview"
                        id={undefined}
            />
</div>,
        ],
        "className": "GistPreview",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <b>
                Error:
</b>,
              " Please enter a valid ISO 8601 date",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Error:",
              },
              "ref": null,
              "rendered": "Error:",
              "type": "b",
            },
            " Please enter a valid ISO 8601 date",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Viewing Gists by 
                defunkt
</h2>,
              <Gist
                className="GistPreview"
                id={undefined}
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Viewing Gists by ",
                  "defunkt",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Viewing Gists by ",
                "defunkt",
              ],
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "GistPreview",
                "id": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="GistPreview"
>
          <p>
                    <b>
                              Error:
                    </b>
                     Please enter a valid ISO 8601 date
          </p>
          <DefaultMessage />
          <div>
                    <h2>
                              Viewing Gists by 
                              defunkt
                    </h2>
                    <Gist
                              className="GistPreview"
                              id={undefined}
                    />
          </div>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              <b>
                            Error:
              </b>
               Please enter a valid ISO 8601 date
</p>,
            <DefaultMessage />,
            <div>
              <h2>
                            Viewing Gists by 
                            defunkt
              </h2>
              <Gist
                            className="GistPreview"
                            id={undefined}
              />
</div>,
          ],
          "className": "GistPreview",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <b>
                  Error:
</b>,
                " Please enter a valid ISO 8601 date",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Error:",
                },
                "ref": null,
                "rendered": "Error:",
                "type": "b",
              },
              " Please enter a valid ISO 8601 date",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2>
                  Viewing Gists by 
                  defunkt
</h2>,
                <Gist
                  className="GistPreview"
                  id={undefined}
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Viewing Gists by ",
                    "defunkt",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Viewing Gists by ",
                  "defunkt",
                ],
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "GistPreview",
                  "id": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`defaultMessage 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainContent
    dataFound={true}
    invalidDate={true}
    userName="defunkt"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="GistPreview"
>
        <p>
                <b>
                        Error:
                </b>
                 Please enter a valid ISO 8601 date
        </p>
        <DefaultMessage />
        <div>
                <h2>
                        Viewing Gists by 
                        defunkt
                </h2>
                <Gist
                        className="GistPreview"
                        id={undefined}
                />
        </div>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            <b>
                        Error:
            </b>
             Please enter a valid ISO 8601 date
</p>,
          <DefaultMessage />,
          <div>
            <h2>
                        Viewing Gists by 
                        defunkt
            </h2>
            <Gist
                        className="GistPreview"
                        id={undefined}
            />
</div>,
        ],
        "className": "GistPreview",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <b>
                Error:
</b>,
              " Please enter a valid ISO 8601 date",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Error:",
              },
              "ref": null,
              "rendered": "Error:",
              "type": "b",
            },
            " Please enter a valid ISO 8601 date",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Viewing Gists by 
                defunkt
</h2>,
              <Gist
                className="GistPreview"
                id={undefined}
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Viewing Gists by ",
                  "defunkt",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Viewing Gists by ",
                "defunkt",
              ],
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "GistPreview",
                "id": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="GistPreview"
>
          <p>
                    <b>
                              Error:
                    </b>
                     Please enter a valid ISO 8601 date
          </p>
          <DefaultMessage />
          <div>
                    <h2>
                              Viewing Gists by 
                              defunkt
                    </h2>
                    <Gist
                              className="GistPreview"
                              id={undefined}
                    />
          </div>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              <b>
                            Error:
              </b>
               Please enter a valid ISO 8601 date
</p>,
            <DefaultMessage />,
            <div>
              <h2>
                            Viewing Gists by 
                            defunkt
              </h2>
              <Gist
                            className="GistPreview"
                            id={undefined}
              />
</div>,
          ],
          "className": "GistPreview",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <b>
                  Error:
</b>,
                " Please enter a valid ISO 8601 date",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Error:",
                },
                "ref": null,
                "rendered": "Error:",
                "type": "b",
              },
              " Please enter a valid ISO 8601 date",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2>
                  Viewing Gists by 
                  defunkt
</h2>,
                <Gist
                  className="GistPreview"
                  id={undefined}
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Viewing Gists by ",
                    "defunkt",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Viewing Gists by ",
                  "defunkt",
                ],
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "GistPreview",
                  "id": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`invalidDate 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainContent
    dataFound={true}
    invalidDate={true}
    userName="defunkt"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="GistPreview"
>
        <p>
                <b>
                        Error:
                </b>
                 Please enter a valid ISO 8601 date
        </p>
        <DefaultMessage />
        <div>
                <h2>
                        Viewing Gists by 
                        defunkt
                </h2>
                <Gist
                        className="GistPreview"
                        id={undefined}
                />
        </div>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            <b>
                        Error:
            </b>
             Please enter a valid ISO 8601 date
</p>,
          <DefaultMessage />,
          <div>
            <h2>
                        Viewing Gists by 
                        defunkt
            </h2>
            <Gist
                        className="GistPreview"
                        id={undefined}
            />
</div>,
        ],
        "className": "GistPreview",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <b>
                Error:
</b>,
              " Please enter a valid ISO 8601 date",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Error:",
              },
              "ref": null,
              "rendered": "Error:",
              "type": "b",
            },
            " Please enter a valid ISO 8601 date",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Viewing Gists by 
                defunkt
</h2>,
              <Gist
                className="GistPreview"
                id={undefined}
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Viewing Gists by ",
                  "defunkt",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Viewing Gists by ",
                "defunkt",
              ],
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "GistPreview",
                "id": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="GistPreview"
>
          <p>
                    <b>
                              Error:
                    </b>
                     Please enter a valid ISO 8601 date
          </p>
          <DefaultMessage />
          <div>
                    <h2>
                              Viewing Gists by 
                              defunkt
                    </h2>
                    <Gist
                              className="GistPreview"
                              id={undefined}
                    />
          </div>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              <b>
                            Error:
              </b>
               Please enter a valid ISO 8601 date
</p>,
            <DefaultMessage />,
            <div>
              <h2>
                            Viewing Gists by 
                            defunkt
              </h2>
              <Gist
                            className="GistPreview"
                            id={undefined}
              />
</div>,
          ],
          "className": "GistPreview",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <b>
                  Error:
</b>,
                " Please enter a valid ISO 8601 date",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Error:",
                },
                "ref": null,
                "rendered": "Error:",
                "type": "b",
              },
              " Please enter a valid ISO 8601 date",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2>
                  Viewing Gists by 
                  defunkt
</h2>,
                <Gist
                  className="GistPreview"
                  id={undefined}
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Viewing Gists by ",
                    "defunkt",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Viewing Gists by ",
                  "defunkt",
                ],
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "GistPreview",
                  "id": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
